src/Manager/ShaderManager.cs

```diff
@@ -47,14 +41,20 @@ public static Shader FetchShader(string name)
 		}
 	}
 	
+	private static Dictionary<string, ShaderMaterial> _shaderMaterials = new Dictionary<string, ShaderMaterial>();
+	
 	public static ShaderMaterial FetchShaderMaterial(string name)
 	{
-		var shader = ShaderManager.FetchShader(name);
-		
-		var shaderMaterial = new ShaderMaterial();
-		
-		shaderMaterial.Shader = shader;
-		
-		return shaderMaterial;
+	
+		if (!_shaderMaterials.ContainsKey(name))
+		{
+			var shader = ShaderManager.FetchShader(name);
+			var shaderMaterial = new ShaderMaterial();
+			shaderMaterial.Shader = shader;
+			_shaderMaterials[name] = shaderMaterial;
+		}
```

Comment:
not 100% sure if good idea to create one master and give everyone a duplicate but has almost no overhead

Reply:
Or like this? (Doesn't waste that one ShaderMaterial)
```C#
	public static ShaderMaterial FetchShaderMaterial(string name)
	{
		ShaderMaterial shaderMaterial = null;
	
		if (!_shaderMaterials.ContainsKey(name))
		{
			var shader = ShaderManager.FetchShader(name);
			
			shaderMaterial = new ShaderMaterial();

			shaderMaterial.Shader = shader;
			
			_shaderMaterials[name] = shaderMaterial;
		}
		else
		{
			shaderMaterial = (ShaderMaterial)(_shaderMaterials[name].Duplicate(false));
		}

		return shaderMaterial;
	}
	```